composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/domain/ValidateEmailUseCase.kt

```diff
@@ -0,0 +1,12 @@
+package sp.bvantur.tasky.core.domain
+
+interface ValidateEmailUseCase {
+    operator fun invoke(email: String): Boolean
+}
+
+class ValidateEmailUseCaseImpl : ValidateEmailUseCase {
+    override fun invoke(email: String): Boolean {
+        val emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+"
```

Comment:
Validating a real email is quite more complex than this simple Regex here. There's a built in function to do that:
```
PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
```

Reply:
I am aware of this one and from what I checked it is Android only and it can not be used for KMP projects in a shared source set. I could maybe use the expect-actual mechanism and use it for Android target, but in my opinion, this is overkill for now.

Reply:
Yes sorry, with all those mentoring participants I keep on forgetting that you're working with KMP here 😄

Reply:
No worries. :) Completely understandable. :)